python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -267,18 +267,25 @@ def _update(self):
                     self.load_metrics.mark_active(
                         self.provider.internal_ip(node_id))
                 else:
-                    logger.error(f"StandardAutoscaler: {node_id}: Terminating "
-                                 "failed to setup/initialize node.")
+                    nodes_to_terminate.append(node_id)
+                    self.num_failed_updates[node_id] += 1
+                del self.updaters[node_id]
+            if nodes_to_terminate:
+                # Some nodes in nodes_to_terminate may have been terminated
+                # during an update. Only terminate currently active nodes.
+                active_nodes = self.workers()
+                active_nodes_to_terminate = [
+                    node for node in nodes_to_terminate if node in active_nodes
+                ]
+                for node_id in active_nodes_to_terminate:
```

Comment:
Should we debug log/info log on nodes that are in `nodes_to_terminate` and not in `active_nodes`?

Reply:
Good idea.
That also makes things a bit more symmetric.